src/Nethermind/Nethermind.Consensus/Producers/BlockProducerBase.cs

```diff
@@ -238,7 +238,7 @@ public bool IsProducingBlocks(ulong? maxProducingInterval)
             return Task.FromResult((Block?)null);
         }
 
-        private bool TrySetState(Keccak? parentStateRoot)
+        protected bool TrySetState(Keccak? parentStateRoot)
```

Comment:
needed to be protected?

Reply:
I need to use this on PostMergeBlockProducer.

Reply:
I think the processing shouldn't be done there in the first place.